core/domain/stats_domain.py

```diff
@@ -43,28 +43,32 @@ def total_answer_count(self):
         return sum(self.answers.values())
 
     @classmethod
-    def get_multi(cls, exploration_id, rule_data):
-        """Gets domain objects corresponding to the given rule data.
+    def get_multi(cls, exploration_state_list, rule_str_list):
+        """Gets domain objects given a list of exploration and state tuples.
 
         Args:
-            exploration_id: the exploration id
-            rule_data: a list of dicts, each with the following keys:
-                (state_name, rule_str).
+            exploration_state_list: a list of exploration ID and state name
+                tuples
+            rule_str_list: a list of rule spec strings which are used to filter
+                the answers matched to the provided explorations and states
+
+        Returns a list with a list of StateRuleAnswerLog objects for each group
+        of answers matched to each exploration ID-state name.
         """
         # TODO(sll): Should each rule_str be unicode instead?
-        answer_log_models = (
+        answer_log_models_list = (
             stats_models.StateRuleAnswerLogModel.get_or_create_multi(
-                exploration_id, rule_data))
-        return [cls(answer_log_model.answers)
-                for answer_log_model in answer_log_models]
+                exploration_state_list, rule_str_list))
+        return [
+            [cls(answer_log_model.answers)
```

Comment:
Does this inner list have the same number of elements as len(rule_str_list), in the same order? If so, might be worth mentioning that somewhere.

(Though I thought the idea was to just use the rule_strs as filters -- not as individual keys -- and concat everything to get a big log of answers for the given state that match any of the rule filters? Otherwise you're effectively doing a query on the 3-tuple (exp_id, state_name, rule_str).)